crates/matrix-sdk-common/src/ring_buffer.rs

```diff
@@ -405,4 +413,33 @@ mod tests {
         // Then only the last N items remain
         assert_eq!(ring_buffer.iter().map(String::as_str).collect::<Vec<_>>(), vec!["6", "7"]);
     }
+
+    #[test]
+    fn test_retain() {
+        let mut ring_buffer = RingBuffer::new(2);
+        ring_buffer.push(1);
+        ring_buffer.push(2);
+
+        ring_buffer.retain(|v| v % 2 == 0);
+
+        assert_eq!(ring_buffer.len(), 1);
+        assert_eq!(ring_buffer.get(0).copied().unwrap(), 2);
+    }
+
+    #[test]
+    fn test_default_isnt_zero_capacity() {
+        let mut ring_buffer = RingBuffer::default();
```

Comment:
Is it really useful to have a `default` impl? Who would want a `RingBuffer` with unknown, surprisingly-non-zero capacity?

Reply:
I've added this comment to https://github.com/matrix-org/matrix-rust-sdk/issues/3017 :-)